test/integration/auth.spec.ts

```diff
@@ -1191,6 +1321,63 @@ describe('admin.auth', () => {
     ]).should.eventually.be.fulfilled;
   });
 
+  describe('deleteUsers()', () => {
+    it('deletes users', async () => {
+      const uid1 = await admin.auth().createUser({}).then((ur) => ur.uid);
+      const uid2 = await admin.auth().createUser({}).then((ur) => ur.uid);
+      const uid3 = await admin.auth().createUser({}).then((ur) => ur.uid);
+      const ids = [{uid: uid1}, {uid: uid2}, {uid: uid3}];
+
+      return admin.auth().deleteUsers([uid1, uid2, uid3])
+        .then((deleteUsersResult) => {
+          expect(deleteUsersResult.successCount).to.equal(3);
+          expect(deleteUsersResult.failureCount).to.equal(0);
+          expect(deleteUsersResult.errors).to.have.length(0);
+
+          return admin.auth().getUsers(ids);
+        })
+        .then((getUsersResult) => {
+          expect(getUsersResult.users).to.have.length(0);
+          expect(getUsersResult.notFound).to.have.deep.members(ids);
+        });
+    });
+
+    it('deletes users that exist even when non-existing users also specified', async () => {
+      const uid1 = await admin.auth().createUser({}).then((ur) => ur.uid);
+      const uid2 = 'uid-that-doesnt-exist';
+      const ids = [{uid: uid1}, {uid: uid2}];
+
+      return admin.auth().deleteUsers([uid1, uid2])
+        .then((deleteUsersResult) => {
+          expect(deleteUsersResult.successCount).to.equal(2);
+          expect(deleteUsersResult.failureCount).to.equal(0);
+          expect(deleteUsersResult.errors).to.have.length(0);
+
+          return admin.auth().getUsers(ids);
+        })
+        .then((getUsersResult) => {
+          expect(getUsersResult.users).to.have.length(0);
+          expect(getUsersResult.notFound).to.have.deep.members(ids);
+        });
+    });
+
+    it('is idempotent', async () => {
```

Comment:
Was this the decided upon behavior?

Reply:
Yes; see the clarifications section of the api proposal: "DeleteUsers() call that specifies only users that don’t exist (or no users at all) results in an empty result set rather than an error. This allows developers to not need to special case this situation, as well as makes this call idempotent. (The returned SuccessCount value would indicate the number of users that no longer exist, regardless of their initial state, i.e. it would also include users that already did not exist.)"

If you wanted to push back on that, we still could; we haven't fully committed to this behaviour yet. (fwiw, I prefer it as is.) In order to go down that route, we'd have to make a backend change too though, since the backend also behaves this way and doesn't provide any indication as to whether an account was deleted or never existed in the first place.

Reply:
I see. My only concern is that it's different from how the existing `deleteUser()` API works. But I also think the behavior proposed here is actually better/convenient for bulk deletes.